projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -759,16 +772,151 @@ function wpcom_add_new_page_check( $post_id, $post ) {
 }
 add_action( 'wp_insert_post', 'wpcom_add_new_page_check', 10, 3 );
 
+/**
+ * Determine if the site has an 'About' page.
+ * We do this by loading the `en` annotation for the site and theme.
+ * We then check if there is a post with the title 'About'. If there is, we get the `hs_old_id` and check to make sure a corresponding post exists in the user's locale.
+ *
+ * @return int|null The page ID of the 'About' page if it exists, null otherwise.
+ */
+function wpcom_get_site_about_page_id() {
+	// First, attempt to get the page ID from the cache.
+	$blog_id                 = get_current_blog_id();
+	$about_page_id_cache_key = "wpcom_get_site_about_page_id_{$blog_id}";
+	$about_page_id           = wp_cache_get( $about_page_id_cache_key, 'wpcom' );
+	if ( false !== $about_page_id ) {
+		return $about_page_id;
+	}
+
+	// If we don't have a cached page id yet, go through the process of looking it up.
+	$annotation = wpcom_get_theme_annotation( get_stylesheet() );
+
+	// Return null if there is no annotation or the annotation doesn't have any content.
+	if ( ! $annotation || ! isset( $annotation['content'] ) || ! is_array( $annotation['content'] ) ) {
+		return null;
+	}
+
+	// Use the annotation to build up a list of 'About' pages.
+	$headstart_about_pages = array_filter(
+		$annotation['content'],
+		function ( $page ) {
+			if ( 'page' !== $page['post_type'] ) {
+				return false;
+			}
+
+			if ( 'about' === $page['post_name'] || false !== strpos( $page['post_title'], 'About' ) ) {
+				return true;
+			}
+		}
+	);
+
+	// Return null if there are no 'About' pages.
+	if ( empty( $headstart_about_pages ) ) {
+		return null;
+	}
+
+	// Get the hs_old_ids for the 'About' pages. We'll use these to find any published pages with the same hs_old_id.
+	$headstart_about_page_hs_old_ids = array_map(
+		function ( $about_page ) {
+			return $about_page['hs_old_id'];
+		},
+		$headstart_about_pages
+	);
+
+	// Return null if there aren't any hs_old_ids to check.
+	if ( empty( $headstart_about_page_hs_old_ids ) ) {
+		return null;
+	}
+
+	$filters     = array(
+		'post_type'    => 'page',
+		'numberposts'  => 1,
+		'orderby'      => 'meta_value',
+		'meta_key'     => '_hs_old_id',
+		'meta_compare' => 'IN',
+		'meta_value'   => $headstart_about_page_hs_old_ids,
+	);
+	$about_pages = get_posts( $filters );
+
+	// Return null if we couldn't find any published pages matching the hs_old_ids.
+	if ( empty( $about_pages ) ) {
+		return null;
+	}
+
+	// Cache to hs_old_ids so we can validate the post being updated later.
+	wp_cache_set( "headstart_about_page_hs_old_ids_{$blog_id}", $headstart_about_page_hs_old_ids, 'wpcom' );
+
+	// Cache the about page id so we don't have to look it up from scratch every time.
+	wp_cache_set( $about_page_id_cache_key, $about_pages[0]->ID, 'wpcom' );
```

Comment:
I have a concern here around the caching -- as things stand, we're only going to cache the value when we find a valid about page, but we have a number of `return null` exit conditions above.

To keep things simple, I think it may be easiest if you move the identification logic to a separate function, which would look like the following:
```php
function wpcom_get_site_about_page_id() {
	// First, attempt to get the page ID from the cache.
	$blog_id                 = get_current_blog_id();
	$about_page_id_cache_key = "wpcom_get_site_about_page_id_{$blog_id}";
	$about_page_id           = wp_cache_get( $about_page_id_cache_key, 'jetpack_mu_wpcom' );

	if ( false !== $about_page_id ) {
		// We should decide how we want to cache cases where we don't have an about page.
		// I've used -1 for now, but we could also use 0. I think `null` might run into implementation details in the caching implementation.
		$about_page_id = (int) $about_page_id;
		if ( $about_page_id > 0 ) {
			return $about_page_id;
		}
		return null;
	}

	// Call the current function implementation without any caching logic.
	$about_page_id = wpcom_find_site_about_page_id(); // could also be wpcom_identify_site_about_page_id() or similar

	if ( null === $about_page_id ) {
		$value_to_cache = -1;
	} else {
		$value_to_cache = $about_page_id;
	}

	// Use wp_cache_add() to avoid cache stampedes, and specify an expiration time - I've used an hour here.
	wp_cache_add( $about_page_id_cache_key, $value_to_cache, 'jetpack_mu_wpcom', 3600 );

	return $about_page_id;
}
```